In [16]:
# --------- Importing NumPy as Alias 'np' --------- #
import numpy as np

# --------- Importing Pandas as Alias 'pd' --------- #
import pandas as pd

# --------- Importing Matplotlib as Alias 'plt' --------- #
import matplotlib.pyplot as plt

# --------- Importing Matplotlib Ticker as Alias 'ticker' --------- #
import matplotlib.ticker as ticker

# --------- Importing patheffects from Matplotlib --------- #
from matplotlib import patheffects

# --------- Importing Rectangle from Matplotlib --------- #
from matplotlib.patches import Rectangle

# --------- Importing Matplotlib colors as Alias 'mcolors' --------- #
import matplotlib.colors as mcolors

# --------- Importing subplots from Plotly --------- #
from plotly.subplots import make_subplots

# --------- Importing Plotly as Alis 'px' --------- #
import plotly.express as px

# --------- Importing Graph Objects as Alias 'go' --------- #
import plotly.graph_objects as go

# --------- Importing Seaborn as Alias 'sns' --------- #
import seaborn as sns

# --------- Importing StatsModel as Alias 'sm' --------- #
import statsmodels.api as sm

# --------- Importing Regular Expression as Alias 're' --------- #
import re

# --------- Importing Stats as Alias 'stats' --------- #
from scipy import stats

# --------- Importing Hstack From Scipy Sparse --------- #
from scipy.sparse import hstack

# --------- Importing CSR Matrix From Scipy Sparse --------- #
from scipy.sparse import csr_matrix

# --------- Importing Ordinal Encoder From Sklearn Preprocessing --------- #
from sklearn.preprocessing import OrdinalEncoder

# ----------- Importing Term Frequency–Inverse Document Frequency Vectoriser From Sklearn Feature Extraction ---------- #
from sklearn.feature_extraction.text import TfidfVectorizer

# ------------ Importing KMeans From Sklearn Cluster ----------- #
from sklearn.cluster import KMeans

# ------------ Importing Sillhouette Score From Sklearn Metrics ----------- #
from sklearn.metrics import silhouette_score

# ------------ Importing Davies Bouldin Score Score From Sklearn Metrics ----------- #
from sklearn.metrics import calinski_harabasz_score

# ------------ Importing Gaussian Mixture From Sklearn Mixture ----------- #
from sklearn.mixture import GaussianMixture

# ------------ Importing Gaussian Mixture From Sklearn Mixture ----------- #
from nltk.corpus import stopwords

# ------------ Importing Gaussian Mixture From Sklearn Mixture ----------- #
from nltk.stem import WordNetLemmatizer

# ------------ Importing Gaussian Mixture From Sklearn Mixture ----------- #
import nltk

# ------------ Downloading NLTK Data ------------ #
nltk.download('stopwords')
nltk.download('wordnet')

# ------------ Importing Warnings ------------ #
import warnings

# ------------ Filtering Warnings ------------ #
warnings.filterwarnings("ignore")

# --------- Importing Median Absolute Deviation From Statsmodels --------- #
from statsmodels.robust.scale import mad

# ------------ Importing Train-Test-Split From Sklearn Model Selection ----------- #
from sklearn.model_selection import train_test_split

# ------------ Importing Random Forest Regressor From Sklearn Ensemble ----------- #
from sklearn.ensemble import RandomForestRegressor

# ------------ Importing Multi Output Regressor From Sklearn Multi-Output ----------- #
from sklearn.multioutput import MultiOutputRegressor

# ------------ Importing RandomizedSearchCV, KFold, Cross Validate, Cross Validation Score And ShuffleSplit From Sklearn Model Selection ----------- #
from sklearn.model_selection import RandomizedSearchCV, KFold, cross_validate, cross_val_score, ShuffleSplit

# ------------ Importing Pipeline From Sklearn Pipeline ----------- #
from sklearn.pipeline import Pipeline

# ------------ Importing Plot Tree From Sklearn Tree ----------- #
from sklearn.tree import plot_tree

# ------------ Importing Learning Curve From Sklearn Model Selection ----------- #
from sklearn.model_selection import learning_curve

# ------------ Importing Metrics From Sklearn Metrics ----------- #
from sklearn.metrics import make_scorer, r2_score, mean_absolute_error, mean_squared_error, mean_absolute_percentage_error, explained_variance_score

# ------------ Confirmation of Modules Importation ------------ #
print("Modules Successfully Imported")

Modules Successfully Imported


[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\kunmi\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package wordnet to
[nltk_data]     C:\Users\kunmi\AppData\Roaming\nltk_data...
[nltk_data]   Package wordnet is already up-to-date!


<p>With the output of 'Modules Successfully Imported', I am able to verify that the modules have been imported and I am able to begin Data Preparation and Cleaning</p>

<hr/>
<h2>2.2 Data Overview</h2>

<p>To make sense of the dataset provided, I will firstly need to view each dataset individually and verify the individual contents within the dataframe, its data's presence and also the possibility of merging dataframes.</p>

<hr/>
<h3>2.2.1 First Dataset - Provided Dataset</h3>

<p>The Data Overview will be done on the first dataset in this section. Some operations which will be perfromed are, showing dataset's content, checking length of dataset, checking for missing values, etc.</p>

<hr/>
<h4>2.2.1.1 First Dataset's Content</h4>

<p>I will be importing the first dataset and subsequently view the first 5 data columns of the dataset.</p>

In [24]:
# ----------- Importing Dataset 1 & Assigning to Alias 'df1' ----------- #
df1 = pd.read_csv('../data/raw/graduate employment survey modified.csv')

# ----------- Viewing df1's First 5 Data Rows ----------- #
df1.head(5).style.background_gradient(cmap="Blues")

,year,university,school,degree,employment_rate_overall,employment_rate_ft_perm,basic_monthly_mean,basic_monthly_median,gross_monthly_mean,gross_monthly_median,gross_mthly_25_percentile,gross_mthly_75_percentile,gross_monthly_mean.1,gross_monthly_median.1,gross_mthly_25_percentile.1,gross_mthly_75_percentile.1
0,2013,Nanyang Technological University,College of Business (Nanyang Business School),Accountancy and Business,97.4,96.1,3701,3200,3727,3350,2900,4000,3727,3350,2900,4000
1,2013,Nanyang Technological University,College of Business (Nanyang Business School),Accountancy (3-yr direct Honours Programme),97.1,95.7,2850,2700,2938,2700,2700,2900,2938,2700,2700,2900
2,2013,Nanyang Technological University,College of Business (Nanyang Business School),Business (3-yr direct Honours Programme),90.9,85.7,3053,3000,3214,3000,2700,3500,3214,3000,2700,3500
3,2013,Nanyang Technological University,College of Business (Nanyang Business School),Business and Computing,87.5,87.5,3557,3400,3615,3400,3000,4100,3615,3400,3000,4100
4,2013,Nanyang Technological University,College of Engineering,Aerospace Engineering,95.3,95.3,3494,3500,3536,3500,3100,3816,3536,3500,3100,3816


<p>From the dataframe above, we am able to see that the dataframe contains a total of 16 data columns. I can also conclude that there are a total of 1 Cateogircal Data Columns, 2 Textual Data and 13 Numerical Data Columns. A detailed report is listed below.</p>
<br/>
<p><strong><u>Unique Identifier</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Categorical Data</u></strong></p>
<ul>
    <li>univeristy</li>
</ul>
<p><strong><u>Numerical Data</u></strong></p>
<ul>
    <li>year</li>
    <li>employment_rate_overall	</li>
    <li>employment_rate_ft_perm</li>
    <li>basic_monthly_mean</li>
    <li>basic_monthly_median</li>
    <li>gross_monthly_mean</li>
    <li>gross_monthly_median</li>
    <li>gross_mthly_25_percentile</li>
    <li>gross_mthly_75_percentile</li>
    <li>gross_monthly_mean</li>
    <li>gross_monthly_median</li>
    <li>gross_mthly_25_percentile</li>
    <li>gross_mthly_75_percentile</li>
    <li>gross_monthly_mean.1 (DUPLICATED)</li>
    <li>gross_monthly_median.1 (DUPLICATED)</li>
    <li>gross_mthly_25_percentile.1 (DUPLICATED)</li>
    <li>gross_mthly_75_percentile.1 (DUPLICATED)</li>
</ul>
<p><strong><u>Textual Data</u></strong></p>
<ul>
    <li>school</li>
    <li>degree</li></ul>
<br/>
<p>With the observations stated above, I will be moving on to see the total data rows within the dataset.</p>

<hr/>
<h4>2.2.1.2 First Dataset's Shape</h4>

<p>We will be viewing the shape of the dataframe to see how many data columns there are to fully comprehend the size and scale of the dataset provided to me. This will also allow for identification of potential cross-referencing where the dataset matches each other in terms for size and scale, allowing for merging.</p>

In [28]:
# ----------- df1's Shape ----------- #
shape_data = {'Attribute': ['Rows', 'Columns'], 'Count': list(df1.shape)}

# ----------- Putting Shape Into DataFrame ----------- #
df1_shape = pd.DataFrame(shape_data)

# ----------- Displaying DataFrame ----------- #
df1_shape.style.background_gradient(cmap="Blues")

,Attribute,Count
0,Rows,5135
1,Columns,16


<p>From the output above, I am able to verify that there is a total of 5135 data rows within the first dataset. This shows that I am working with a large dataset with multiple data rows. This also means that the Data Analysis done will be more accurate and relevant considering the wide range of data and information within the dataset. This will make the project much more applicable to the real-life context of the job market.</p>

<hr/>
<h4>2.2.1.3 First Dataset's Missing Data</h4>

<p>We will be checking for missing data within the dataset. This will allow me to plan ahead on whether there is a need to impute missing data to fill up the gaps within these data columns.</p>

In [32]:
# ----------- Placing Missing Values of df1 Into A Dataframe ----------- #
df1MissingData = pd.DataFrame({'Missing Values': df1.isnull().sum().values}, index = df1.columns)

# ----------- Function for Colour Coding ----------- #
def colour_code(values):
    return 'background-color: green' if values == 0 else 'background-color: red'

# ----------- Calling Function for Colour Coding ----------- #
df1MissingData = df1MissingData.style.map(colour_code, subset = ['Missing Values'])

# ----------- Displaying Missing Values ----------- #
df1MissingData

,Missing Values
year,0
university,0
school,0
degree,0
employment_rate_overall,108
employment_rate_ft_perm,108
basic_monthly_mean,108
basic_monthly_median,108
gross_monthly_mean,108
gross_monthly_median,108


<p>From the output above, I am able to make the following observations.</p>
<br/>
<p><strong><u>Data Column with Missing Data</u></strong></p>
<ul>
    <li>employment_rate_overall (108 Missing Data)</li>
    <li>employment_rate_ft_perm (108 Missing Data)</li>
    <li>basic_monthly_mean (108 Missing Data)</li>
    <li>basic_monthly_median (108 Missing Data)</li>
    <li>gross_monthly_mean (108 Missing Data)</li>
    <li>gross_monthly_median (108 Missing Data)</li>
    <li>gross_mthly_25_percentile (108 Missing Data)</li>
    <li>gross_mthly_75_percentile (108 Missing Data)</li>
    <li>gross_monthly_mean.1 (108 Missing Data)</li>
    <li>gross_monthly_median.1 (108 Missing Data)</li>
    <li>gross_mthly_25_percentile.1 (108 Missing Data)</li>
    <li>gross_mthly_75_percentile.1 (108 Missing Data)</li>
</ul>
<br/>
<p>From the observations above, I can conclude that some data columns may require me to impute the missing data. We are also able to derive the potential reasoning for the missing data which could be due to the salary not being stated in the first place which prevents the calculations of all the other data columns.</p>

<hr/>
<h3>2.2.2 Second Dataset</h3>

<p>The Data Overview will be done on the second dataset in this section. Some operations which will be perfromed are, showing dataset's content, checking length of dataset, checking for missing values, etc.</p>

<hr/>
<h4>2.2.2.1 Second Dataset's Content</h4>

<p>I will be importing the second dataset and subsequently view the first 5 data columns of the dataset.</p>

In [38]:
# ----------- Importing Dataset 2 & Assigning to Alias 'df2' ----------- #
df2 = pd.read_csv('../data/raw/labour_force_1_overall_resident.csv')

# ----------- Viewing df2's First 10 Data Rows ----------- #
df2.head(10).style.background_gradient(cmap="Blues")

,year,overall_labour_force,resident_labour_force,Unnamed: 3,Unnamed: 4,Unnamed: 5
0,1991,1673700.000000,1372900.000000,nan,nan,nan
1,1992,1733600.000000,1409900.000000,nan,nan,nan
2,1993,1762700.000000,1421700.000000,nan,nan,nan
3,1994,1842200.000000,1456100.000000,nan,nan,nan
4,1996,2024900.000000,1511500.000000,nan,nan,nan
5,1997,2116000.000000,1538300.000000,nan,nan,nan
6,1998,2187900.000000,1546500.000000,nan,nan,nan
7,1999,2208700.000000,1595900.000000,nan,nan,nan
8,2001,2330500.000000,1644300.000000,nan,nan,nan
9,2002,2320600.000000,1667900.000000,nan,nan,nan


<p>From the dataframe above, I am able to see that there is 1 unique identifier for each of the data rows. I am also able to see that the dataframe contains a total of 3 data columns (inclusive of the UIDs). I can also conclude that there are a total of 1 textual data column and 1 numerical data column. A detailed report is listed below.</p>
<br/>
<p><strong><u>Unique Identifier</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Categorical Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Numerical Data</u></strong></p>
<ul>
    <li>year</li>
    <li>overall_labour_force</li>
    <li>resident_labour_force</li>
</ul>
<p><strong><u>Textual Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Invalid Data</u></strong></p>
<ul>
    <li>Unnamed: 3</li>
    <li>Unnamed: 4</li>
    <li>Unnamed: 5</li>
</ul>
<br/>
<p>With the observations stated above, I will be moving on to see the total data rows within the dataset.</p>

<hr/>
<h4>2.2.2.2 Second Dataset's Shape</h4>

<p>I will be viewing the shape of the dataframe to see how many data columns there are to fully comprehend the size and scale of the dataset provided to me. This will also allow for cross-referencing where I am able to see if all the dataset matches each other in terms for size and scale.</p>

In [42]:
# ----------- df4's Shape ----------- #
shape_data = {'Attribute': ['Rows', 'Columns'], 'Count': list(df2.shape)}

# ----------- Putting Shape Into DataFrame ----------- #
df2_shape = pd.DataFrame(shape_data)

# ----------- Displaying DataFrame ----------- #
df2_shape.style.background_gradient(cmap="Blues")

,Attribute,Count
0,Rows,32
1,Columns,6


<p>From the output above, I am able to verify that there is a total of 123849 data rows within the first dataset. This shows that I am working with a large dataset with multiple data rows. This also means that the Data Analysis done will be more accurate and relevant considering the wide range of data and information within the dataset. This will make the project much more applicable to the real-life context of the job market.</p>

<hr/>
<h4>2.2.2.3 Second Dataset's Missing Data</h4>

<p>I will be checking for missing data within the dataset. This will allow me to plan ahead on whether there is a need to impute missing data to fill up the gaps within these data columns.</p>

In [46]:
# ----------- Placing Missing Values of df2 Into A Dataframe ----------- #
df2MissingData = pd.DataFrame({'Missing Values': df2.isnull().sum().values}, index = df2.columns)

# ----------- Function for Colour Coding ----------- #
def colour_code(values):
    return 'background-color: green' if values == 0 else 'background-color: red'

# ----------- Calling Function for Colour Coding ----------- #
df2MissingData = df2MissingData.style.map(colour_code, subset = ['Missing Values'])

# ----------- Displaying Missing Values ----------- #
df2MissingData

,Missing Values
year,1
overall_labour_force,1
resident_labour_force,1
Unnamed: 3,32
Unnamed: 4,32
Unnamed: 5,32


<p>From the output above, I am able to make the following observations.</p>
<br/>
<p><strong><u>Data Column with Missing Data</u></strong></p>
<ul>
    <li>year (1 Missing Data)</li>
    <li>overall_labour_force (1 Missing Data)</li>
    <li>resident_labour_force (1 Missing Data)</li>
    <li>Unnamed: 3 (32 Missing Data)</li>
    <li>Unnamed: 4 (32 Missing Data)</li>
    <li>Unnamed: 5 (32 Missing Data)</li>
</ul>
<br/>
<p>From the observations above, I can conclude that some data columns may require me to impute and/or remove the missing data.</p>

<hr/>
<h3>2.2.3 Third Dataset</h3>

<p>The Data Overview will be done on the third dataset in this section. Some operations which will be perfromed are, showing dataset's content, checking length of dataset, checking for missing values, etc.</p>

<hr/>
<h4>2.2.3.1 Third Dataset's Content</h4>

<p>I will be importing the third dataset and subsequently view the first 5 data columns of the dataset.</p>

In [52]:
# ----------- Importing Dataset 3 & Assigning to Alias 'df3' ----------- #
df3 = pd.read_csv('../data/raw/res_labour_force_1_edu_sex.csv')

# ----------- Viewing df3's First 5 Data Rows ----------- #
df3.head(5).style.background_gradient(cmap="Blues")

,year,sex,highest_qualification,labour_force
0,1991,male,primary and below,281300
1,1991,male,lower secondary,158800
2,1991,male,secondary,224400
3,1991,male,post-secondary (non-tertiary),76000
4,1991,male,diploma and professional qualification,58300


<p>From the dataframe above, I am able to see that there is 2 unique identifier for each of the data rows. I am also able to see that the dataframe contains a total of 7 data columns (inclusive of the UIDs). I can also conclude that there are a total of 2 categorical data column and 3 numerical data column. A detailed report is listed below.</p>
<br/>
<p><strong><u>Unique Identifier</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Categorical Data</u></strong></p>
<ul>
    <li>sex</li>
    <li>highest_qualification</li>
</ul>
<p><strong><u>Numerical Data</u></strong></p>
<ul>
    <li>year</li>
    <li>labour_force</li>
</ul>
<p><strong><u>Textual Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>With the observations stated above, I will be moving on to see the total data rows within the dataset.</p>

<hr/>
<h4>2.2.3.2 Third Dataset's Shape</h4>

<p>I will be viewing the shape of the dataframe to see how many data columns there are to fully comprehend the size and scale of the dataset provided to me. This will also allow for cross-referencing where I am able to see if all the dataset matches each other in terms for size and scale.</p>

In [56]:
# ----------- df4's Shape ----------- #
shape_data = {'Attribute': ['Rows', 'Columns'], 'Count': list(df3.shape)}

# ----------- Putting Shape Into DataFrame ----------- #
df3_shape = pd.DataFrame(shape_data)

# ----------- Displaying DataFrame ----------- #
df3_shape.style.background_gradient(cmap="Blues")

,Attribute,Count
0,Rows,372
1,Columns,4


<p>From the output above, I am able to verify that there is a total of 372 data rows within the first dataset. This shows that I am working with a large dataset with multiple data rows. This also means that the Data Analysis done will be more accurate and relevant considering the wide range of data and information within the dataset. This will make the project much more applicable to the real-life context of the job market.</p>

<hr/>
<h4>2.2.3.3 Third Dataset's Missing Data</h4>

<p>I will be checking for missing data within the dataset. This will allow me to plan ahead on whether there is a need to impute missing data to fill up the gaps within these data columns.</p>

In [60]:
# ----------- Placing Missing Values of df3 Into A Dataframe ----------- #
df3MissingData = pd.DataFrame({'Missing Values': df3.isnull().sum().values}, index = df3.columns)

# ----------- Function for Colour Coding ----------- #
def colour_code(values):
    return 'background-color: green' if values == 0 else 'background-color: red'

# ----------- Calling Function for Colour Coding ----------- #
df3MissingData = df3MissingData.style.map(colour_code, subset = ['Missing Values'])

# ----------- Displaying Missing Values ----------- #
df3MissingData

,Missing Values
year,0
sex,0
highest_qualification,0
labour_force,0


<p>From the output above, I am able to make the following observations.</p>
<br/>
<p><strong><u>Data Column with Missing Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>From the observations above, I can conclude that some data columns may require me to impute the missing data.</p>

<hr/>
<h3>2.2.4 Fourth Dataset</h3>

<p>The Data Overview will be done on the fourth dataset in this section. Some operations which will be perfromed are, showing dataset's content, checking length of dataset, checking for missing values, etc.</p>

<hr/>
<h4>2.2.4.1 Fourth Dataset's Content</h4>

<p>I will be importing the fourth dataset and subsequently view the first 5 data columns of the dataset.</p>

In [66]:
# ----------- Importing Dataset 4 & Assigning to Alias 'df4' ----------- #
df4 = pd.read_csv('../data/raw/res_labour_force_2_age_sex.csv')

# ----------- Viewing df4's First 5 Data Rows ----------- #
df4.head(5).style.background_gradient(cmap="Blues")

,year,sex,age,resident_labour_force
0,1991,male,15-19,36900
1,1991,male,20-24,99400
2,1991,male,25-29,134500
3,1991,male,30-34,143300
4,1991,male,35-39,131000


<p>From the dataframe above, I am able to see that there is 3 unique identifiers (UIDs) within the dataset. I am also able to see that the dataframe contains a total of 18 data columns (inclusive of UIDs). I can also conclude that there are a total of 10 categorical data column and 4 numerical data column (exclusive of UID) and 1 Textual Data. A detailed report is listed below.</p>
<br/>
<p><strong><u>Unique Identifier</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Categorical Data</u></strong></p>
<ul>
    <li>sex</li>
    <li>age</li>
</li>
</ul>
<p><strong><u>Numerical Data</u></strong></p>
<ul>
    <li>resident_labour_force</li>
</ul>
<p><strong><u>Textual Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>With the observations stated above, I will be moving on to see the total data rows within the dataset.</p>

<hr/>
<h4>2.2.4.2 Fourth Dataset's Shape</h4>

<p>I will be viewing the shape of the dataframe to see how many data columns there are to fully comprehend the size and scale of the dataset provided to me. This will also allow for cross-referencing where I am able to see if all the dataset matches each other in terms for size and scale.</p>

In [70]:
# ----------- df4's Shape ----------- #
shape_data = {'Attribute': ['Rows', 'Columns'], 'Count': list(df4.shape)}

# ----------- Putting Shape Into DataFrame ----------- #
df4_shape = pd.DataFrame(shape_data)

# ----------- Displaying DataFrame ----------- #
df4_shape.style.background_gradient(cmap="Blues")

,Attribute,Count
0,Rows,744
1,Columns,4


<p>From the output above, I am able to verify that there is a total of 744 data rows within the first dataset. This shows that I am working with a large dataset with multiple data rows. This also means that the Data Analysis done will be more accurate and relevant considering the wide range of data and information within the dataset. This will make the project much more applicable to the real-life context of the job market.</p>

<hr/>
<h4>2.2.4.3 Fourth Dataset's Missing Data</h4>

<p>I will be checking for missing data within the dataset. This will allow me to plan ahead on whether there is a need to impute missing data to fill up the gaps within these data columns.</p>

In [74]:
# ----------- Placing Missing Values of df4 Into A Dataframe ----------- #
df4MissingData = pd.DataFrame({'Missing Values': df4.isnull().sum().values}, index = df4.columns)

# ----------- Function for Colour Coding ----------- #
def colour_code(values):
    return 'background-color: green' if values == 0 else 'background-color: red'

# ----------- Calling Function for Colour Coding ----------- #
df4MissingData = df4MissingData.style.map(colour_code, subset = ['Missing Values'])

# ----------- Displaying Missing Values ----------- #
df4MissingData

,Missing Values
year,0
sex,0
age,0
resident_labour_force,0


<p>From the output above, I am able to make the following observations.</p>
<br/>
<p><strong><u>Data Column with Missing Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>From the observations above, I can conclude that there are no Missing Data that needs to be Imputed.</p>

<hr/>
<h3>2.2.5 Fifth Dataset</h3>

<p>The Data Overview will be done on the fifth dataset in this section. Some operations which will be perfromed are, showing dataset's content, checking length of dataset, checking for missing values, etc.</p>

<hr/>
<h4>2.2.5.1 Fifth Dataset's Content</h4>

<p>I will be importing the fifth dataset and subsequently view the first 5 data rows of the dataset.</p>

In [80]:
# ----------- Importing Dataset 5 & Assigning to Alias 'df5' ----------- #
df5 = pd.read_csv('../data/raw/res_LFPR_1_sex_agg_age.csv')

# ----------- Viewing df5's First 5 Data Rows ----------- #
df5.head(5).style.background_gradient(cmap="Blues")

,year,sex,age,resident_labour_force_participation_rate
0,1991,male,25-64,92.600000
1,1991,male,15-24,55.900000
2,1991,male,25-54,97.500000
3,1991,male,55-64,59.700000
4,1991,male,65 & Over,19.900000


<p>From the dataframe above, I am able to see that there is 1 unique identifier (UID) within the dataset. I am also able to see that the dataframe contains a total of 5 data columns (inclusive of UIDs). I can also conclude that there are a total of 2 categorical data column and 2 miscellaneous data columns (exclusive of UID). A detailed report is listed below.</p>
<br/>
<p><strong><u>Unique Identifier</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Categorical Data</u></strong></p>
<ul>
    <li>sex</li>
    <li>age</li>
</ul>
<p><strong><u>Numerical Data</u></strong></p>
<ul>
    <li>resident_labour_force_participation_rate</li>
</ul>
<p><strong><u>Textual Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>With the observations stated above, I will be moving on to see the total data rows within the dataset.</p>

<hr/>
<h4>2.2.5.2 Fifth Dataset's Shape</h4>

<p>I will be viewing the shape of the dataframe to see how many data columns there are to fully comprehend the size and scale of the dataset provided to me. This will also allow for cross-referencing where I am able to see if all the dataset matches each other in terms for size and scale.</p>

In [84]:
# ----------- df5's Shape ----------- #
shape_data = {'Attribute': ['Rows', 'Columns'], 'Count': list(df5.shape)}

# ----------- Putting Shape Into DataFrame ----------- #
df5_shape = pd.DataFrame(shape_data)

# ----------- Displaying DataFrame ----------- #
df5_shape.style.background_gradient(cmap="Blues")

,Attribute,Count
0,Rows,310
1,Columns,4


<p>From the output above, I am able to verify that there is a total of 310 data rows within the fifth dataset. This shows that I am working with a large dataset with multiple data rows. This also means that the Data Analysis done will be more accurate and relevant considering the wide range of data and information within the dataset. This will make the project much more applicable to the real-life context of the job market.</p>

<hr/>
<h4>2.2.5.3 Fifth Dataset's Missing Data</h4>

<p>I will be checking for missing data within the dataset. This will allow me to plan ahead on whether there is a need to impute missing data to fill up the gaps within these data columns.</p>

In [88]:
# ----------- Placing Missing Values of df5 Into A Dataframe ----------- #
df5MissingData = pd.DataFrame({'Missing Values': df5.isnull().sum().values}, index = df5.columns)

# ----------- Function for Colour Coding ----------- #
def colour_code(values):
    return 'background-color: green' if values == 0 else 'background-color: red'

# ----------- Calling Function for Colour Coding ----------- #
df5MissingData = df5MissingData.style.map(colour_code, subset = ['Missing Values'])

# ----------- Displaying Missing Values ----------- #
df5MissingData

,Missing Values
year,0
sex,0
age,0
resident_labour_force_participation_rate,0


<p>From the output above, I am able to make the following observations.</p>
<br/>
<p><strong><u>Data Column with Missing Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>From the observations above, I can conclude that there are no Missing Data that needs to be Imputed.</p>

<hr/>
<h3>2.2.6 Sixth Dataset</h3>

<p>The Data Overview will be done on the sixth dataset in this section. Some operations which will be perfromed are, showing dataset's content, checking length of dataset, checking for missing values, etc.</p>

<hr/>
<h4>2.2.6.1 Sixth Dataset's Content</h4>

<p>I will be importing the sixth dataset and subsequently view the first 5 data rows of the dataset.</p>

In [94]:
# ----------- Importing Dataset 6 & Assigning to Alias 'df6' ----------- #
df6 = pd.read_excel('../data/raw/Median Salary Per Industry.xlsx')

# ----------- Viewing df5's First 5 Data Rows ----------- #
df6.head(5).style.background_gradient(cmap="Blues")

,Amt,Gender,Industry,Year
0,6105,Male,C Manufacturing,2023
1,5070,Male,F Construction,2023
2,5387,Male,G-U Services,2023
3,5168,Male,G Wholesale & Retail Trade,2023
4,3435,Male,H Transportation & Storage,2023


<p>From the dataframe above, we are able to see that the dataframe contains a total of 4 Data Columns. I can also conclude that there are a total of 2 Categorical Data Column and 2 Numerical Data Columns. A detailed report is listed below.</p>
<br/>
<p><strong><u>Unique Identifier</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Categorical Data</u></strong></p>
<ul>
    <li>Gender</li>
    <li>Industry</li>
</ul>
<p><strong><u>Numerical Data</u></strong></p>
<ul>
    <li>Amt</li>
    <li>Year</li>
</ul>
<p><strong><u>Textual Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>With the observations stated above, I will be moving on to see the total Data Rows within the dataset.</p>

<hr/>
<h4>2.2.6.2 Sixth Dataset's Shape</h4>

<p>I will be viewing the shape of the dataframe to see how many data columns there are to fully comprehend the size and scale of the dataset provided to me. This will also allow for cross-referencing where I am able to see if all the dataset matches each other in terms for size and scale.</p>

In [98]:
# ----------- df6's Shape ----------- #
shape_data = {'Attribute': ['Rows', 'Columns'], 'Count': list(df6.shape)}

# ----------- Putting Shape Into DataFrame ----------- #
df6_shape = pd.DataFrame(shape_data)

# ----------- Displaying DataFrame ----------- #
df6_shape.style.background_gradient(cmap="Blues")

,Attribute,Count
0,Rows,160
1,Columns,4


<p>From the output above, I am able to verify that there is a total of 160 data rows within the sixth dataset. This shows that I am working with a large dataset with multiple data rows. This also means that the Data Analysis done will be more accurate and relevant considering the wide range of data and information within the dataset. This will make the project much more applicable to the real-life context of the job market.</p>

<hr/>
<h4>2.2.6.3 Sixth Dataset's Missing Data</h4>

<p>I will be checking for missing data within the dataset. This will allow me to plan ahead on whether there is a need to impute missing data to fill up the gaps within these data columns.</p>

In [102]:
# ----------- Placing Missing Values of df6 Into A Dataframe ----------- #
df6MissingData = pd.DataFrame({'Missing Values': df6.isnull().sum().values}, index = df6.columns)

# ----------- Function for Colour Coding ----------- #
def colour_code(values):
    return 'background-color: green' if values == 0 else 'background-color: red'

# ----------- Calling Function for Colour Coding ----------- #
df6MissingData = df6MissingData.style.map(colour_code, subset = ['Missing Values'])

# ----------- Displaying Missing Values ----------- #
df6MissingData

,Missing Values
Amt,0
Gender,0
Industry,0
Year,0


<p>From the output above, I am able to make the following observations.</p>
<br/>
<p><strong><u>Data Column with Missing Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>From the observations above, I can conclude that there are no Missing Data that needs to be Imputed.</p>

<hr/>
<h3>2.2.7 Seventh Dataset</h3>

<p>The Data Overview will be done on the seventh dataset in this section. Some operations which will be perfromed are, showing dataset's content, checking length of dataset, checking for missing values, etc.</p>

<hr/>
<h4>2.2.7.1 Seventh Dataset's Content</h4>

<p>I will be importing the seventh dataset and subsequently view the first 5 data rows of the dataset.</p>

In [108]:
# ----------- Importing Dataset 7 & Assigning to Alias 'df7' ----------- #
df7 = pd.read_csv('../data/raw/Population.csv')

# ----------- Viewing df5's First 5 Data Rows ----------- #
df7.head(5).style.background_gradient(cmap="Blues")

,Year,2024,2023,2022,2021,2020,2019,2018,2017,2016,2015,2014,2013,2012,2011,2010,2009,2008,2007,2006,2005,2004,2003,2002,2001,2000,1999,1998,1997,1996,1995,1994,1993,1992,1991,1990,1989,1988,1987,1986,1985,1984,1983,1982,1981,1980,1979,1978,1977,1976,1975,1974,1973,1972,1971,1970,1969,1968,1967,1966,1965,1964,1963,1962,1961,1960,1959,1958,1957,1956,1955,1954,1953,1952,1951,1950
0,Total Population (Number),"6,036,860","5,917,648","5,637,022","5,453,566","5,685,807","5,703,569","5,638,676","5,612,253","5,607,283","5,535,002","5,469,724","5,399,162","5,312,437","5,183,688","5,076,732","4,987,573","4,839,396","4,588,599","4,401,365","4,265,762","4,166,664","4,114,826","4,175,950","4,138,012","4,027,887","3,958,723","3,927,213","3,796,038","3,670,704","3,524,506","3,419,048","3,313,471","3,230,698","3,135,083","3,047,132","2,930,901","2,846,108","2,774,789","2,733,373","2,735,957","2,732,221","2,681,061","2,646,466","2,532,835","2,413,945","2,383,500","2,353,600","2,325,300","2,293,300","2,262,600","2,229,800","2,193,000","2,152,400","2,112,900","2,074,507","2,042,500","2,012,000","1,977,600","1,934,400","1,886,900","1,841,600","1,795,000","1,750,200","1,702,400","1,646,400","1,587,200","1,518,800","1,445,929","1,371,600","1,305,500","1,248,200","1,191,800","1,127,000","1,068,100","1,022,100"


<p>From the dataframe above, we are able to see that the dataframe theoretically contains a total of 2 Data Columns. I can also conclude that there are a total of 2 Numerical Data Columns. A detailed report is listed below.</p>
<br/>
<p><strong><u>Unique Identifier</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Categorical Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<p><strong><u>Numerical Data</u></strong></p>
<ul>
    <li>Year</li>
    <li>Total Population (Number)</li>
</ul>
<p><strong><u>Textual Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>With the observations stated above, I will be moving on to see the total Data Rows within the dataset.</p>

<hr/>
<h4>2.2.7.2 Seventh Dataset's Shape</h4>

<p>I will be viewing the shape of the dataframe to see how many data columns there are to fully comprehend the size and scale of the dataset provided to me. This will also allow for cross-referencing where I am able to see if all the dataset matches each other in terms for size and scale.</p>

In [112]:
# ----------- df7's Shape ----------- #
shape_data = {'Attribute': ['Rows', 'Columns'], 'Count': list(df7.shape)}

# ----------- Putting Shape Into DataFrame ----------- #
df7_shape = pd.DataFrame(shape_data)

# ----------- Displaying DataFrame ----------- #
df7_shape.style.background_gradient(cmap="Blues")

,Attribute,Count
0,Rows,1
1,Columns,76


<p>From the output above, I am able to verify that there is a total of 1 data rows and 76 data column within the sixth dataset. This shows that there is a need to transpose the DataFrame so as to turn the Data Columns into Data Rows and vice versa. This operation will be done after the checking of Missing Data.</p>

<hr/>
<h4>2.2.7.3 Seventh Dataset's Missing Data</h4>

<p>I will be checking for missing data within the dataset. This will allow me to plan ahead on whether there is a need to impute missing data to fill up the gaps within these data columns.</p>

In [116]:
# ----------- Placing Missing Values of df6 Into A Dataframe ----------- #
df7MissingData = pd.DataFrame({'Missing Values': df7.isnull().sum().values}, index = df7.columns)

# ----------- Function for Colour Coding ----------- #
def colour_code(values):
    return 'background-color: green' if values == 0 else 'background-color: red'

# ----------- Calling Function for Colour Coding ----------- #
df7MissingData = df7MissingData.style.map(colour_code, subset = ['Missing Values'])

# ----------- Displaying Missing Values ----------- #
df7MissingData

,Missing Values
Year,0
2024,0
2023,0
2022,0
2021,0
2020,0
2019,0
2018,0
2017,0
2016,0


<p>From the output above, I am able to make the following observations.</p>
<br/>
<p><strong><u>Data Column with Missing Data</u></strong></p>
<ul>
    <li>NONE</li>
</ul>
<br/>
<p>From the observations above, I can conclude that there are no Missing Data that needs to be Imputed.</p>

<hr/>
<h4>2.2.7.4 Seventh Dataset's Transpose Operation</h4>

<p>As previously observed, the Data Columns of Dataset 7 is suppose to be the Data Rows whereas the Data Rows for Dataset 7 is suppose to be the Data Columns. Therefore, to address this issue, we will be performing a Transpose Operation in the cell below and subsequently convert this Dataset into a conventional DataFrame.</p>

In [120]:
# ---------- Transpose of DataFrame ------------ #
df7 = df7.T.reset_index()

# ---------- Rename of Data Columns ---------- #
df7.columns = ["Year", "Population"]

# --------- Removing of First Data Row ----------- #
df7 = df7.iloc[1:]

# ----------- Displaying of DataFrame ------------ #
df7.style.background_gradient(cmap="Blues")

,Year,Population
1,2024,"6,036,860"
2,2023,"5,917,648"
3,2022,"5,637,022"
4,2021,"5,453,566"
5,2020,"5,685,807"
6,2019,"5,703,569"
7,2018,"5,638,676"
8,2017,"5,612,253"
9,2016,"5,607,283"
10,2015,"5,535,002"


<p>With reference to the output of the DataFrame above, we are able to see that the transpose operation have been carried out successfully and we are able to proceed to handle Duplicated Data within all the datasets.</p>

<hr/>
<h2>2.4 Handling Duplicated Data</h2>